test/org/sagebionetworks/bridge/services/ScheduledActivityServiceMockTest.java

```diff
@@ -372,111 +380,154 @@ public void deleteActivitiesForSchedulePlanRejectsBadValue() {
+        
+        verify(activityDao).saveActivities(scheduledActivityListCaptor.capture());
+        List<ScheduledActivity> saves = scheduledActivityListCaptor.getValue();
+        assertActivityGuids(saves, "AAA");
     }
     
     @Test
     public void persistedAndScheduledIncludedInResultsV3() {
-        List<ScheduledActivity> scheduled = createNewActivities("CCC");
-        List<ScheduledActivity> db = createStartedActivities("CCC");
-
-        List<ScheduledActivity> saves = merge(scheduled, db, V3_MERGE);
-        assertActivities(saves);
+        SchedulePlan aaa = schedulePlan("AAA");
+        SchedulePlan bbb = schedulePlan("BBB");
+        SchedulePlan ccc = schedulePlan("CCC");
+        
+        when(schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY)).thenReturn(Lists.newArrayList(aaa,bbb,ccc));
+        when(activityDao.getActivities(eq(TIME_ZONE), any())).thenReturn(createStartedActivities("CCC"+TIME_PORTION));
+        
+        List<ScheduledActivity> returnedActivities = service.getScheduledActivities(createScheduleContext(NOW).build());
+        assertNotNull(returnedActivities.get(0).getStartedOn());
+        assertActivityGuids(returnedActivities, "AAA", "BBB", "CCC");
         
-        scheduled = service.orderActivities(scheduled, V3_FILTER);
-        assertNotNull(scheduled.get(0).getStartedOn());
+        verify(activityDao).saveActivities(scheduledActivityListCaptor.capture());
+        List<ScheduledActivity> saves = scheduledActivityListCaptor.getValue();
+        assertActivityGuids(saves, "AAA", "BBB");
     }
 
     @Test
     public void expiredTasksExcludedFromCalculationsV3() {
-        // create activities in the past that are now expired.
-        List<ScheduledActivity> scheduled = createExpiredActivities("AAA","BBB");
-        List<ScheduledActivity> db = createExpiredActivities("AAA","CCC");
+        SchedulePlan aaa = schedulePlan("AAA");
+        SchedulePlan bbb = schedulePlan("BBB");
+        SchedulePlan ccc = schedulePlan("CCC");
+        
+        when(schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY)).thenReturn(Lists.newArrayList(aaa,bbb,ccc));
+        when(activityDao.getActivities(eq(TIME_ZONE), any())).thenReturn(createExpiredActivities("AAA"+TIME_PORTION,"CCC"+TIME_PORTION));
         
-        List<ScheduledActivity> saves = merge(scheduled, db, V3_MERGE);
-        assertActivities(saves);
+        List<ScheduledActivity> returnedActivities = service.getScheduledActivities(createScheduleContext(NOW).build());
+        assertActivityGuids(returnedActivities, "BBB");
         
-        scheduled = service.orderActivities(scheduled, V3_FILTER);
-        assertTrue(scheduled.isEmpty());
+        verify(activityDao).saveActivities(scheduledActivityListCaptor.capture());
+        List<ScheduledActivity> saves = scheduledActivityListCaptor.getValue();
+        assertActivityGuids(saves, "BBB");
```

Comment:
I'm not sure how this test is workable in the new format. The purpose of this test appears to be 3 fold:

1. AAA isn't replaced by the expired task from DDB.
2. BBB isn't saved to DDB because it's expired.
3. CCC isn't added to scheduled (though that's always the case in V3).

It looks like case 1 is preserved with AAA. I'm not sure case 3 is meaningful, but it should be trivial to also preserve this test case.

The only questionable one is case 2. Does the scheduler ever return expired tasks? Or is this a facetious test case?